backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala

```diff
@@ -1133,6 +1137,14 @@ trait StandardAsyncExecutionActor
     Future.successful(FailedNonRetryableExecutionHandle(exception, returnCode, None))
   }
 
+  /**
+    * A latch that says the actor has entered an aborting state. This is used to differentiate kills requested
+    * via Cromwell versus kills due to OOM, etc. Volatile as this actor mixes use of the "synchronous" mailbox
+    * with asynchronous Scala futures that may read the value from another thread.
+    */
+  @volatile
+  private var aborting = false
+
```

Comment:
This is a sin against Akka. 🙈

Using this volatile `var` across threads a) works and b) provides a very small diff for reviewers.

But Akka actor best practices say this type of mutation should be stored as (usually immutable) object instances that are replaced as the actor transitions from one "state" to another. I'm more than happy to re-implement this change in a less-Scala-future-and-more-Akka-mailbox way or have someone on the team re-implement it. But I believe the wiring will be a bigger PR for reviewers, which isn't a bad thing, just an FYI.

Reply:
Update: Thanks to #6912 re-ordering the `tryReturnCodeAsInt` matches the `private var aborting` is no longer needed! 🎉